moshi/src/main/java/com/squareup/moshi/internal/Util.java

```diff
@@ -567,17 +565,35 @@ public static String typeAnnotatedWithAnnotations(Type type,
     throw new IllegalStateException("No defaults constructor found for " + targetClass);
   }
 
-  public static JsonDataException missingProperty(String property, JsonReader reader) {
```

Comment:
Because we call this from generated code we cannot delete it!

Reply:
We can’t guarantee that all generated code is generated with the same version, or that the generated code matches the runtime version. Upgrading the runtime version shouldn’t be dangerous.

Reply:
would you rather just inline these directly?

Reply:
It creates a lot of code. I would just re-add this overload, pass `null` for the `jsonName`, and then add a null check so the first branch of the conditional is taken.

Reply:
Just to note - this method was only added since the last release in #900 and could be safely deleted if we want. That's why I was fine making the change here

Reply:
Sounds good. We should set up japicmp...